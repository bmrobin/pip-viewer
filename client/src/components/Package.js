import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const Package = ({ name, version, uninstall }) => (
  <tr>
    <td>{name}</td>
    <td>{version}</td>
    <td>
      <Button bssize="sm" onClick={() => uninstall(name)}>
        X
      </Button>
    </td>
  </tr>
);

Package.propTypes = {
  name: PropTypes.string.isRequired,
  version: PropTypes.string.isRequired,
  uninstall: PropTypes.func.isRequired,
};

export default Package;
