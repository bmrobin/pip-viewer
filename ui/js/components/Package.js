import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Modal } from 'react-bootstrap';

export default function Package({ name, version, uninstall }) {
  const [showWarning, setShowWarning] = useState(false);
  const [uninstallError, setUninstallError] = useState(false);
  const [uninstallErrorMsg, setUninstallErrorMsg] = useState('');

  const handleShowWarning = (show) => {
    setShowWarning(show);
    if (!show) {
      setUninstallErrorMsg('');
      setUninstallError(false);
    }
  };

  const handleUninstall = async () => {
    const responseBody = await uninstall(name);
    if (responseBody.status !== 200) {
      setUninstallError(true);
      const response = await responseBody.json();
      setUninstallErrorMsg(response);
    } else {
      handleShowWarning(false);
      setUninstallErrorMsg('');
      setUninstallError(false);
    }
  };

  return (
    <tr>
      <td>{name}</td>
      <td>{version}</td>
      <td>
        <Button bssize="sm" onClick={() => handleShowWarning(true)}>
          X
        </Button>

        <Modal show={showWarning} onHide={() => handleShowWarning(false)}>
          <Modal.Header closeButton>
            <Modal.Title>Uninstall {name}=={version}</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            {
              !uninstallError &&
              'Are you sure you want to uninstall this package?'
            }
            {
              uninstallError &&
              <div className="error">
                <p>Error uninstalling &quot;{name}&quot;</p>
                <p>{uninstallErrorMsg}</p>
              </div>
            }
          </Modal.Body>
          <Modal.Footer>
            <Button variant="secondary" onClick={() => handleShowWarning(false)}>
              No
            </Button>
            <Button variant="primary" onClick={handleUninstall}>
              Yes
            </Button>
          </Modal.Footer>
        </Modal>

      </td>
    </tr>
  );
}

Package.propTypes = {
  name: PropTypes.string.isRequired,
  version: PropTypes.string.isRequired,
  uninstall: PropTypes.func.isRequired,
};
