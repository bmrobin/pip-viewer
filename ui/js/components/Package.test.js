import React from 'react';
import renderer from 'react-test-renderer';
import Package from './Package';

describe('Package component', () => {
  it('should render component', () => {
    const component = renderer.create(<Package name="pkgName" version="1.2.3" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
