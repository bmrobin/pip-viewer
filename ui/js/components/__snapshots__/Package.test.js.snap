// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package component should render component 1`] = `
<tr>
  <td>
    pkgName
  </td>
  <td>
    1.2.3
  </td>
  <td>
    <button
      bssize="sm"
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </td>
</tr>
`;
